### PR TITLE
resourceQuota: optimize Equals of ResourceLists

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
optimize ResourceList Equals function: reduce a loop.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/quota/resources.go [hunk 1]
```diff
@@ -36,20 +40,16 @@ func Equals(a api.ResourceList, b api.ResourceList) bool {
...
 	}
-	for key, value1 := range b {
-		value2, found := a[key]
-		if !found {
-			return false
-		}
-		if value1.Cmp(value2) != 0 {
-			return false
-		}
-	}
```
Comment (zjj2wry) [inline pkg/quota/resources.go]: why remove this, if a contains b, will alse return true

Comment (hzxuzhonghu) [reply pkg/quota/resources.go]: a contains b (len(a) > len(b)) does not return true.

Comment (hzxuzhonghu) [reply pkg/quota/resources.go]: It verifies if a and b equals

Comment (zjj2wry) [reply pkg/quota/resources.go]: lgtm

Comment (wgliang) [reply pkg/quota/resources.go]: Determine the key is equal to determine the value is equal? I think the definition of equality should be key equal and the value is equal. So need call value1.Cmp(value2),I think.

Comment (hzxuzhonghu) [reply pkg/quota/resources.go]: Yes, and that's what it does now.

Comment (wgliang) [reply pkg/quota/resources.go]: U removed 
```
if value1.Cmp(value2) != 0 {	
	return false	
}
```

Comment (hzxuzhonghu) [reply pkg/quota/resources.go]: Oh, That's a big bug, will revert it.

Comment (hzxuzhonghu) [reply pkg/quota/resources.go]: No, it is hidden. see https://github.com/kubernetes/kubernetes/pull/64688/files/7feaec1008274263593495b279ee20809a6cfbf5#diff-2c2419fccdeffe90dff794bba3251a5dL35

Comment (wgliang) [reply pkg/quota/resources.go]: Oh, I see.IGNORE.

Comment (CaoShuFeng) [reply pkg/quota/resources.go]: looks good to me

**Pull request merged**